### PR TITLE
Optimize the GUI front-end project structure

### PR DESCRIPTION
**Pull request by m1m1sha**
Can `pnpm` be used to replace `yarn`

### CODE REVIEW

**Comment by KKRainbow** [comment]
what advantages do pnpm have over yarn

**Comment by KKRainbow** [comment]
I have no preference between using yarn or pnpm. Use pnpm is okay for me if necessary.

**Comment by m1m1sha** [comment]
> I have no preference between using yarn or pnpm. Use pnpm is okay for me if necessary.

i just providing an opinion, using pnpm or yarn is also acceptable for me
pnpm seems to be faster than yarn and can avoid duplicate package installations, but it doesn't seem to be fully compatible with yarn
﻿
im not particularly familiar with pr, so i may be a bit presumptuous in all aspects. if you have any questions, please point them out. this is also an opportunity for me to learn

thanks~

**Comment by KKRainbow** [comment]
In my opinion it would be better to replace the package mgmt system in a new PR, because that change may require some modification in workflow and also make the PR too huge to review.

**Comment by m1m1sha** [comment]
> In my opinion it would be better to replace the package mgmt system in a new PR, because that change may require some modification in github action workflow and also make the PR too huge to review.

okk, got it

**Comment by m1m1sha** [comment]
i don't know how to make `vue office (volar)` use the formatting rules of `eslint`, so there are some slight differences between `vue office`'s formatting of `vue` files and `eslint` in the current pr

**Comment by KKRainbow** [comment]
> i don't know how to make `vue office (volar)` use the formatting rules of `eslint`, so there are some slight differences between `vue office`'s formatting of `vue` files and `eslint` in the current pr

I'm not very familiar with vue things, but it looks like a minor and acceptable problem. may resolve it later.

**Comment by m1m1sha** [comment]
> > i don't know how to make  use the formatting rules of , so there are some slight differences between 's formatting of  files and  in the current pr`vue office (volar)``eslint``vue office``vue``eslint`
> 
> I'm not very familiar with vue things, but it looks like a minor and acceptable problem. may resolve it later.

i have resolved it by using the workspace of vscode

**Comment by m1m1sha** [comment]
@KKRainbow do you go through codacy static code analysis when submitting the front-end code?  i just split the code from the original files into their respective folders, and except for i18n loading logic, did not change the logic of other code. but it reported 95 issues.

although i think going through codacy review is a good thing, but i don't know how to use codacy to prompt me to modify the issue in local, there are no eslint errors or warnings in my local.  so i think it might be a bit troublesome, but I will try to modify it

**Comment by KKRainbow** [comment]
> @KKRainbow do you go through codacy static code analysis when submitting the front-end code? i just split the code from the original files into their respective folders, and except for i18n loading logic, did not change the logic of other code. but it reported 95 issues.
> 
> although i think going through codacy review is a good thing, but i don't know how to use codacy to prompt me to modify the issue in local, there are no eslint errors or warnings in my local. so i think it might be a bit troublesome, but I will try to modify it

No, you can just ignore what Codacy is saying. I'm planning to remove this check because it doesn't provide enough useful warnings.

**Comment by KKRainbow** [comment]
It need some time for me to review this change and I'll do that after work.

**Comment by m1m1sha** [comment]
> It need some time for me to review this change and I'll do that after work.

okk

File: EasyTier.code-workspace [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+	"folders": [
+		{
+			"path": "."
```
Comment (KKRainbow) [inline EasyTier.code-workspace:4]: 如果 vscode 直接打开的是 easytier 项目的根目录，这个文件会影响 vscode 的行为么

File: easytier-gui/.gitignore [hunk 2]
```diff
@@ -13,8 +13,9 @@ dist-ssr
 *.local
 
 # Editor directories and files
-.vscode/*
+.vscode/
```
Comment (KKRainbow) [inline easytier-gui/.gitignore]: 这里保持原有的 ./vscode/* 似乎会好些？

File: EasyTier.code-workspace [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+		},
+		{
+			"path": "easytier-gui"
+		},
+		{
+			"path": "easytier"
+		}
+	],
+	"settings": {}
+}
```
Comment (KKRainbow) [inline EasyTier.code-workspace:87]: 文件末尾留空行

File: easytier-gui/.npmrc [hunk 4]
```diff
@@ -0,0 +1,2 @@
+shamefully-hoist=true
+strict-peer-dependencies=false
```
Comment (KKRainbow) [inline easytier-gui/.npmrc]: 文件末尾留空行

Comment (KKRainbow) [reply easytier-gui/.npmrc]: easytier-gui 依赖 yarn，也需要 npmrc 么

File: easytier-gui/.vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,74 @@
+{
+  "i18n-ally.localesPaths": [
+    "locales"
+  ],
+  // Enable the ESlint flat config support
```
Comment (KKRainbow) [inline easytier-gui/.vscode/settings.json]: json 文件应该是不支持注释的，可以去掉注释

File: easytier-gui/src/auto-imports.d.ts
Comment (KKRainbow) [inline easytier-gui/src/auto-imports.d.ts:1]: 这个文件是如何生成的，后续如何保持这个文件是最新的呢

File: easytier-gui/src/main.ts [hunk 6]
```diff
@@ -1,362 +1,43 @@
...
-
-const app = createApp(App);
+app.use(router)
+app.use(createPinia())
 app.use(i18n, { useScope: 'global' })
-app.use(pinia)
-app.use(PrimeVue);
-app.use(ToastService);
-app.mount("#app");
+loadLanguageAsync(localStorage.getItem('lang') || 'en')
```
Comment (KKRainbow) [inline easytier-gui/src/main.ts]: 这个 async function 可以直接这么调用么，需要 await 不

File: easytier-gui/src/modules/i18n.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    .map(([path, loadLocale]) => [path.match(/([\w-]*)\.yml$/)?.[1], loadLocale]),
+) as Record<Locale, () => Promise<{ default: Record<string, string> }>>
+
+export const availableLocales = Object.keys(localesMap)
+
+const loadedLanguages: string[] = []
+
+function setI18nLanguage(lang: Locale) {
+  i18n.global.locale.value = lang as any
+  localStorage.setItem('locale', lang)
```
Comment (KKRainbow) [inline easytier-gui/src/modules/i18n.ts]: 这里是否需要改成 'lang'

File: EasyTier.code-workspace (hunk 1 above)
Comment (m1m1sha) [reply EasyTier.code-workspace:4]: 并不会，如果直接打开 easytier 根目录是和未添加工作区时是一样的
添加工作区是因为 easytier 的前端项目是作为子项目嵌套在文件夹内，有些插件无法正确发现嵌套文件夹项目，从而导致无法正常运行

File: easytier-gui/src/auto-imports.d.ts
Comment (m1m1sha) [reply easytier-gui/src/auto-imports.d.ts:1]: 这个是通过 `vite` 插件 `unplugin-auto-import` 自动生成的
后续启动前端项目时会自动生成，启动项目时修改前端文件也会自动更新

File: easytier-gui/.npmrc (hunk 4 above)
Comment (m1m1sha) [reply easytier-gui/.npmrc]: > easytier-gui 依赖 yarn，也需要 npmrc 么

yarn 使用 yarnrc.yml ，这个我先移除吧

**Comment by m1m1sha** [comment]
eslint 配置项中 ignores 未生效，暂时使用命令行参数代替
使用 @eslint/config-inspector 能够解析发现 ignores 但是就是不生效，不知道是什么原因

**Comment by KKRainbow** [comment]
main.ts 中直接调用 async 函数会执行么，那个是不是无效语句了成

**Comment by m1m1sha** [comment]
> main.ts 中直接调用 async 函数会执行么，那个是不是无效语句了成

确实有些欠考虑，async 函数会执行但不会阻塞，如果函数运行时间比较长的话，有可能会导致 vue 挂载完成后还没执行完毕。但是在 async 函数内部的异步函数会正常阻塞直到完成后再进行下一步，我修改一下哈

File: easytier/src/common/config.rs
Comment (KKRainbow) [inline easytier/src/common/config.rs:1]: 感觉好像没有 rebase 完全，这个文件的修改是 main 分支上的

Comment (KKRainbow) [reply easytier/src/common/config.rs:1]: 后面几个文件也是

**Review by KKRainbow — CHANGES_REQUESTED** [review]

File: easytier/src/common/config.rs
Comment (m1m1sha) [reply easytier/src/common/config.rs:1]: 我看一下

Comment (m1m1sha) [reply easytier/src/common/config.rs:1]: 应该是因为我网络问题，修改没有完全更新全 :(
感觉我一直在重复合并冲突。。。

**Comment by m1m1sha** [comment]
要不我重开一个pr？污染git记录了

**Comment by KKRainbow** [comment]
> 要不我重开一个pr？污染git记录了

没事，你本地整好 force push 上来应该就行了

**Review by KKRainbow — APPROVED** [review]

**Pull request merged**